include/libtorrent/add_torrent_params.hpp

```diff
@@ -313,7 +315,7 @@ namespace libtorrent
 		// can be set to control the initial file priorities when adding a
 		// torrent. The semantics are the same as for
 		// ``torrent_handle::prioritize_files()``.
-		std::vector<std::uint8_t> file_priorities;
+		vector<std::uint8_t, file_index_t> file_priorities;
```

Comment:
I think it's probably best to not expose ``libtorrent::vector<>``, but the index types are trivially castable to integers (and with deprecated-functions enabled, implicitly convertible). There's no way to define converters in swig? i.e. piece_index_t <-> int etc?